store/tikv/region_cache.go

```diff
@@ -69,71 +69,71 @@ type Region struct {
 	lastAccess int64          // last region access time, see checkRegionCacheTTL
 }
 
+// EngineCnt presents max engine type supported by TiDB.
+const EngineCnt = 2
+
 // RegionStore represents region stores info
 // it will be store as unsafe.Pointer and be load at once
 type RegionStore struct {
-	workTiKVIdx    int32    // point to current work peer in meta.Peers and work store in stores(same idx) for tikv peer
-	workTiFlashIdx int32    // point to current work peer in meta.Peers and work store in stores(same idx) for tiflash peer
-	stores         []*Store // stores in this region
-	storeFails     []uint32 // snapshots of store's fail, need reload when `storeFails[curr] != stores[cur].fail`
+	currIdx    [EngineCnt]int32
+	stores     [EngineCnt][]*Store // stores in this region
+	storeFails [EngineCnt][]uint32 // snapshots of store's fail, need reload when `storeFails[curr] != stores[cur].fail`
 }
 
 // clone clones region store struct.
 func (r *RegionStore) clone() *RegionStore {
-	storeFails := make([]uint32, len(r.stores))
-	for i, e := range r.storeFails {
-		storeFails[i] = e
+	nr := &RegionStore{
+		currIdx: r.currIdx,
 	}
-	return &RegionStore{
-		workTiFlashIdx: r.workTiFlashIdx,
-		workTiKVIdx:    r.workTiKVIdx,
-		stores:         r.stores,
-		storeFails:     storeFails,
+	for engineType := range r.stores {
+		nr.stores[engineType] = make([]*Store, len(r.stores[engineType]))
+		for j, store := range r.stores[engineType] {
+			nr.stores[engineType][j] = store
+		}
 	}
+	for engineType := range r.storeFails {
+		nr.storeFails[engineType] = make([]uint32, len(r.storeFails[engineType]))
+		for j, e := range r.storeFails[engineType] {
+			nr.storeFails[engineType][j] = e
+		}
+	}
+	return nr
 }
 
 // return next follower store's index
-func (r *RegionStore) follower(seed uint32) int32 {
-	l := uint32(len(r.stores))
+func (r *RegionStore) follower(engine kv.StoreType, seed uint32) int32 {
+	l := uint32(len(r.stores[engine]))
 	if l <= 1 {
-		return r.workTiKVIdx
+		return r.currIdx[engine]
 	}
 
 	for retry := l - 1; retry > 0; retry-- {
 		followerIdx := int32(seed % (l - 1))
-		if followerIdx >= r.workTiKVIdx {
+		if followerIdx >= r.currIdx[engine] {
```

Comment:
Why `>=`? Although this logic isn't changed in this PR, I think it's better to comment this logic. It's hard to understand to me.

Reply:
I thinking follower read is confused too, could you refine this in you PR, this one shouldn't change logic and do any fix logic